feat!: upgrade to Angular 20

griest024 opened the pull request:
## PR Checklist

- [ ] Commit message follows our [contributing guidelines](https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit)
- [ ] Tests added/updated (for bug fixes/features)
- [ ] Documentation added/updated (for bug fixes/features)

## PR Type
<!-- Select the relevant type(s) -->

- [ ] Bug fix
- [x] Feature
- [ ] Style update
- [ ] Refactor
- [ ] Test
- [ ] Build
- [ ] CI
- [ ] Docs
- [ ] Performance
- [ ] Other (please describe)

## Current behavior
<!-- Describe the current behavior or link to relevant issue -->

Fixes: #3807

## New behavior
<!-- Describe the changes -->

## Breaking change?

- [x] Yes
- [ ] No

<!-- If yes, describe impact and migration path -->
Upgrade your application to Angular 20
## Additional context
<!-- Any other relevant information -->

damienwebdev commented on libs/authorizenet/tsconfig.lib.json:
```diff
@@ -3,14 +3,14 @@
     "outDir": "../../dist/out-tsc/libs/authorizenet",
     "declarationMap": true,
-    "moduleResolution": "node",
+    "moduleResolution": "bundler",
```
What's this do?

> 'bundler' for use with bundlers. Like node16 and nodenext, this mode supports package.json "imports" and "exports", but unlike the Node.js resolution modes, bundler never requires file extensions on relative paths in imports.

Neat I guess?

griest024 replied:
This was part of an automatic angular migration. I assume this is what we want.

damienwebdev commented on apps/daffio/src/app/core/dynamic-fragment/service.ts:
```diff
@@ -1,7 +1,8 @@
 import {
   Inject,
   Injectable,
+  DOCUMENT,
```
Is `DOCUMENT` in `core` in ng19? Is it still in `common` in ng20?

griest024 replied:
no, `common` <20, `core` >= 20

damienwebdev commented on libs/authorizenet/tsconfig.lib.json:
```diff
@@ -3,14 +3,14 @@
     "outDir": "../../dist/out-tsc/libs/authorizenet",
     "declarationMap": true,
-    "moduleResolution": "node",
+    "moduleResolution": "bundler",
```
Can this change be made in `develop` currently?

griest024 replied:
I'm not sure

damienwebdev commented on libs/design/tsconfig.json:
```diff
@@ -100,7 +100,7 @@
         "libs/design/switch/src"
       ],
-       "@daffodil/design/tabs": [
+			"@daffodil/design/tabs": [
```
Indent looks unecessary

griest024 replied:
This indentation is correct now, I prefer it

damienwebdev commented on libs/driver/hubspot/testing/src/factories/hubspot-response.factory.spec.ts:
```diff
@@ -1,6 +1,6 @@
 import { TestBed } from '@angular/core/testing';
 
-import { HubspotResponse } from '@daffodil/driver/hubspot/models/hubspot-response';
+import { HubspotResponse } from '@daffodil/driver/hubspot';
```
How did you catch this? Love the change, just unsure how you found it.

griest024 replied:
It triggered a build error, likely due to the `bundler` change

damienwebdev commented on libs/product/driver/magento/src/transforms/applied-filter-transformer.service.ts:
```diff
@@ -55,7 +56,7 @@ export class MagentoProductAppliedFiltersTransformService {
    * Returns an array for Equal type and a string for Match type.
    */
-  private getFilterValue(type: DaffFilterType, value: DaffFilterRequest['value']): string | string[] {
+  private getFilterValue(type: DaffFilterType, value: DaffFilterEqualRequest['value']): string | string[] {
```
Can this be extracted to `develop`?

griest024 replied:
yes

griest024 replied:
done 296c10c8fb669dc56735ace2cba34b4272e25b5d

damienwebdev commented on tools/builders/tsconfig.json:
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "outDir": "../../dist/builders",
-    "module": "commonjs",
+    "module": "preserve",
```
What does this do?

griest024 replied:
this tells TS to output ECMAScript as opposed to commonjs. This *should* be what we want but its hard to tell if it will work in 100% of cases

damienwebdev commented:
@griest024 needs rebase

griest024 committed:
feat!: upgrade to Angular 20

BREAKING CHANGE: Angular has been upgraded to v20

`npx ng update @angular/{core,cli,pwa,cdk} @angular-eslint/schematics @angular-devkit/core typescript-eslint @typescript-eslint/{parser,eslint-plugin,util
s,types}@latest @ngrx/schematics @fortawesome/{angular-fontawesome,fontawesome-svg-core,free-brands-svg-icons,free-regular-svg-icons,free-solid-svg-icons} angular-in-memory-web-api apollo-angular @schematics/angular`

The installed Angular CLI version is outdated.
Installing a temporary Angular CLI versioned 20.3.2 to perform the update.
Using package manager: npm
Collecting installed dependencies...
Found 123 dependencies.
Fetching dependency metadata from registry...
    Updating package.json with dependency @angular-devkit/architect @ "0.2003.2" (was "0.1902.12")...
    Updating package.json with dependency @angular-devkit/build-angular @ "20.3.2" (was "19.2.12")...
    Updating package.json with dependency @angular-devkit/core @ "20.3.2" (was "19.2.12")...
    Updating package.json with dependency @angular-devkit/schematics @ "20.3.2" (was "19.2.12")...
    Updating package.json with dependency @angular-eslint/builder @ "20.3.0" (was "19.8.1")...
    Updating package.json with dependency @angular-eslint/eslint-plugin @ "20.3.0" (was "19.8.1")...
    Updating package.json with dependency @angular-eslint/eslint-plugin-template @ "20.3.0" (was "19.8.1")...
    Updating package.json with dependency @angular-eslint/schematics @ "20.3.0" (was "19.8.1")...
    Updating package.json with dependency @angular-eslint/template-parser @ "20.3.0" (was "19.8.1")...
    Updating package.json with dependency @angular/cli @ "20.3.2" (was "19.2.12")...
    Updating package.json with dependency @angular/compiler-cli @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/language-service @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @ngrx/schematics @ "20.0.1" (was "19.0.1")...
    Updating package.json with dependency @ngrx/store-devtools @ "20.0.1" (was "19.0.1")...
    Updating package.json with dependency @typescript-eslint/eslint-plugin @ "8.44.0" (was "8.37.0")...
    Updating package.json with dependency @typescript-eslint/parser @ "8.44.0" (was "8.37.0")...
    Updating package.json with dependency @typescript-eslint/types @ "8.44.0" (was "8.37.0")...
    Updating package.json with dependency @typescript-eslint/utils @ "8.44.0" (was "8.37.0")...
    Updating package.json with dependency angular-eslint @ "20.3.0" (was "19.8.1")...
    Updating package.json with dependency ng-packagr @ "20.3.0" (was "19.0.1")...
    Updating package.json with dependency typescript @ "5.9.2" (was "5.5.4")...
    Updating package.json with dependency typescript-eslint @ "8.44.0" (was "8.37.0")...
    Updating package.json with dependency @angular/animations @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/cdk @ "20.2.4" (was "19.2.16")...
    Updating package.json with dependency @angular/common @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/compiler @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/core @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/elements @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/forms @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/platform-browser @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/platform-browser-dynamic @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/platform-server @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/pwa @ "20.3.2" (was "19.2.12")...
    Updating package.json with dependency @angular/router @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/service-worker @ "20.3.1" (was "19.2.11")...
    Updating package.json with dependency @angular/ssr @ "20.3.2" (was "19.2.12")...
    Updating package.json with dependency @fortawesome/angular-fontawesome @ "3.0.0" (was "1.0.0")...
    Updating package.json with dependency @fortawesome/fontawesome-svg-core @ "7.0.1" (was "6.7.2")...
    Updating package.json with dependency @fortawesome/free-brands-svg-icons @ "7.0.1" (was "6.7.2")...
    Updating package.json with dependency @fortawesome/free-regular-svg-icons @ "7.0.1" (was "6.7.2")...
    Updating package.json with dependency @fortawesome/free-solid-svg-icons @ "7.0.1" (was "6.7.2")...
    Updating package.json with dependency @ngrx/component @ "20.0.1" (was "19.0.1")...
    Updating package.json with dependency @ngrx/effects @ "20.0.1" (was "19.0.1")...
    Updating package.json with dependency @ngrx/entity @ "20.0.1" (was "19.0.1")...
    Updating package.json with dependency @ngrx/operators @ "20.0.1" (was "19.0.1")...
    Updating package.json with dependency @ngrx/router-store @ "20.0.1" (was "19.0.1")...
    Updating package.json with dependency @ngrx/store @ "20.0.1" (was "19.0.1")...
    Updating package.json with dependency angular-in-memory-web-api @ "0.20.0" (was "0.19.0")...
    Updating package.json with dependency apollo-angular @ "11.0.0" (was "10.0.3")...
UPDATE package.json (5218 bytes)
✔ Cleaning node modules directory
✔ Installing packages
** Executing migrations of package '@angular-eslint/schematics' **

❯ Updates @angular-eslint to v20.
UPDATE package.json (5218 bytes)
✔ Packages installed successfully.
  Migration completed (1 file modified).

** Executing migrations of package '@angular/cli' **

❯ Update workspace generation defaults to maintain previous style guide behavior.
UPDATE angular.json (73047 bytes)
  Migration completed (1 file modified).

❯ Migrate imports of 'provideServerRendering' from '@angular/platform-server' to '@angular/ssr'.
UPDATE apps/daffio/src/app/app.server.module.ts (704 bytes)
  Migration completed (1 file modified).

❯ Migrate 'provideServerRendering' to use 'withRoutes', and remove 'provideServerRouting' and 'provideServerRoutesConfig' from '@angular/ssr'.
  Migration completed (No changes made).

❯ Update 'moduleResolution' to 'bundler' in TypeScript configurations.
  You can read more about this, here: https://www.typescriptlang.org/tsconfig/#moduleResolution
UPDATE tsconfig.json (622 bytes)
UPDATE libs/authorizenet/tsconfig.lib.json (719 bytes)
UPDATE libs/product/tsconfig.lib.json (714 bytes)
UPDATE libs/related-products/tsconfig.lib.json (723 bytes)
UPDATE libs/upsell-products/tsconfig.lib.json (722 bytes)
UPDATE libs/product-configurable/tsconfig.lib.json (699 bytes)
UPDATE libs/product-composite/tsconfig.lib.json (696 bytes)
UPDATE libs/cart/tsconfig.lib.json (715 bytes)
UPDATE libs/cart-customer/tsconfig.lib.json (715 bytes)
UPDATE libs/category/tsconfig.lib.json (715 bytes)
UPDATE libs/checkout/tsconfig.lib.json (719 bytes)
UPDATE libs/core/tsconfig.lib.json (715 bytes)
UPDATE libs/driver/tsconfig.lib.json (791 bytes)
UPDATE libs/design/tsconfig.lib.json (658 bytes)
UPDATE libs/branding/tsconfig.lib.json (661 bytes)
UPDATE libs/theme-switch/tsconfig.lib.json (733 bytes)
UPDATE libs/navigation/tsconfig.lib.json (717 bytes)
UPDATE libs/newsletter/tsconfig.lib.json (722 bytes)
UPDATE libs/contact/tsconfig.lib.json (612 bytes)
UPDATE libs/auth/tsconfig.lib.json (711 bytes)
UPDATE libs/paypal/tsconfig.lib.json (713 bytes)
UPDATE libs/search/tsconfig.lib.json (699 bytes)
UPDATE libs/search-product/tsconfig.lib.json (721 bytes)
UPDATE libs/search-product-composite/tsconfig.lib.json (731 bytes)
UPDATE libs/search-product-configurable/tsconfig.lib.json (734 bytes)
UPDATE libs/search-category/tsconfig.lib.json (722 bytes)
UPDATE libs/reviews/tsconfig.lib.json (714 bytes)
UPDATE libs/payment/tsconfig.lib.json (699 bytes)
UPDATE libs/customer/tsconfig.lib.json (699 bytes)
UPDATE libs/customer-auth/tsconfig.lib.json (715 bytes)
UPDATE libs/customer-order/tsconfig.lib.json (699 bytes)
UPDATE libs/customer-store-credit/tsconfig.lib.json (730 bytes)
UPDATE libs/cart-store-credit/tsconfig.lib.json (726 bytes)
UPDATE libs/customer-payment/tsconfig.lib.json (699 bytes)
UPDATE libs/customer-payment-authorizenet/tsconfig.lib.json (699 bytes)
  Migration completed (35 files modified).

❯ Remove any karma configuration files that only contain the default content.
  The default configuration is automatically available without a specific project file.
  Migration completed (No changes made).

** Optional migrations of package '@angular/cli' **

This package has 1 optional migration that can be executed.
Optional migrations may be skipped and executed after the update process, if preferred.

 Select the migrations that you'd like to run [use-application-builder] Migrate application projects to the new build system. (https://angular.dev/tools/cli/build-system-migration)

❯ Migrate application projects to the new build system.
  Application projects that are using the '@angular-devkit/build-angular' package's 'browser' and/or 'browser-esbuild' builders will be migrated to use the new 'application' builder.
  You can read more about this, including known issues and limitations, here: https://angular.dev/tools/cli/build-system-migration
  Migration completed (No changes made).

** Executing migrations of package 'angular-eslint' **

❯ Updates @angular-eslint to v20.
UPDATE package.json (5218 bytes)
✔ Packages installed successfully.
  Migration completed (1 file modified).

** Executing migrations of package '@angular/cdk' **

❯ Updates the Angular CDK to v20.

      ✓  Updated Angular CDK to version 20

  Migration completed (No changes made).

** Executing migrations of package '@angular/core' **

❯ Moves imports of `DOCUMENT` from `@angular/common` to `@angular/core`.
UPDATE libs/authorizenet/src/services/accept-js-loading.service.ts (1145 bytes)
UPDATE libs/design/src/core/theming/services/os-theme/ostheme.service.ts (1802 bytes)
UPDATE libs/design/src/core/theming/services/storage/theme-storage.service.ts (2715 bytes)
UPDATE libs/design/src/core/theming/services/class-setter/theme-class-setter.service.ts (2019 bytes)
UPDATE libs/design/src/core/focus/stack.service.ts (1620 bytes)
UPDATE libs/design/src/core/sticky/sticky-tracker.directive.ts (5008 bytes)
UPDATE libs/design/sidebar/src/sidebar/sidebar.component.ts (6022 bytes)
UPDATE libs/design/sidebar/src/sidebar-viewport/scroll-token/scroll.token.ts (1538 bytes)
UPDATE libs/external-router/driver/magento/2.4.3/src/magento.service.ts (1527 bytes)
UPDATE apps/daffio/src/app/core/dynamic-fragment/service.ts (1679 bytes)
UPDATE apps/daffio/src/app/docs/components/scroll-to-top/scroll-to-top.component.ts (1230 bytes)
UPDATE libs/seo/src/canonical/restoreable.service.ts (1802 bytes)
UPDATE libs/seo/src/schema/schema.service.ts (1760 bytes)
UPDATE libs/design/tree/src/tree-item/tree-item.directive.ts (3435 bytes)
UPDATE libs/analytics/driver/data-layer/data-layer.service.ts (863 bytes)
UPDATE libs/product-composite/routing/src/services/query-param.service.ts (674 bytes)
UPDATE libs/core/external-script/src/external-script.service.ts (2994 bytes)
UPDATE libs/design/select/src/select/select.component.ts (9574 bytes)
UPDATE libs/newsletter/driver/hubspot/src/token/hubspot-forms.token.ts (1086 bytes)
UPDATE libs/contact/driver/hubspot/src/token/hubspot-forms.token.ts (1069 bytes)
UPDATE libs/paypal/routing/src/effects/express-redirect.effects.ts (1228 bytes)
UPDATE libs/analytics/driver/data-layer/data-layer.service.spec.ts (1043 bytes)
UPDATE libs/driver/hubspot/src/hubspot-forms.provider.spec.ts (1260 bytes)
UPDATE libs/driver/hubspot/src/hubspot-forms.service.spec.ts (5104 bytes)
UPDATE libs/design/src/core/theming/services/class-setter/theme-class-setter.service.spec.ts (1787 bytes)
UPDATE libs/design/modal/src/specs/focus-management.spec.ts (3125 bytes)
UPDATE libs/design/select/src/select/select.component.spec.ts (9136 bytes)
UPDATE libs/design/sidebar/src/sidebar-viewport/specs/destroy.spec.ts (1527 bytes)
UPDATE libs/seo/src/canonical/restoreable.service.spec.ts (3539 bytes)
UPDATE libs/seo/integration-tests/state/provided-canonical-updates.spec.ts (5841 bytes)
  Migration completed (30 files modified).

❯ Replaces usages of the deprecated InjectFlags enum.
  Migration completed (No changes made).

❯ Replaces usages of the deprecated TestBed.get method with TestBed.inject.
  Migration completed (No changes made).

❯ Adds `BootstrapContext` to `bootstrapApplication` calls in `main.server.ts` to support server rendering.
  Migration completed (No changes made).

** Optional migrations of package '@angular/core' **

This package has 2 optional migrations that can be executed.
Optional migrations may be skipped and executed after the update process, if preferred.

 Select the migrations that you'd like to run [router-current-navigation] Replaces usages of the deprecated Router.getCurrentNavigation method with the Router.currentNavigation signal.

❯ Replaces usages of the deprecated Router.getCurrentNavigation method with the Router.currentNavigation signal.
  Migration completed (No changes made).

** Executing migrations of package '@ngrx/operators' **

❯ Replace deprecated tapResponse signature.
  Migration completed (No changes made).

griest024 committed:
fix schematic breaking changes

adds options for specifying the path for important files. The files that `ng new` generates changed in v20

damienwebdev commented:
Looks like https://github.com/graycoreio/daffodil/blob/develop/libs/theme-switch/tsconfig.lib.json has an older version of `tsconfig`. TBH I'm more curious about how we haven't noticed this yet than anything else.

griest024 committed:
fix theme switch tsconfig

damienwebdev reviewed (approved):
Lgtm
